include/swift/AST/Evaluator.h

```diff
@@ -211,10 +223,11 @@ class Evaluator {
   /// Evaluate the given request and produce its result,
   /// consulting/populating the cache as required.
   template<typename Request>
-  typename Request::OutputType operator()(const Request &request) {
+  llvm:Expected<typename Request::OutputType>
+  operator()(const Request &request) {
     // Check for a cycle.
     if (checkDependency(request))
-      return request.breakCycle();
+      return make_error<RequestErrorInfo<Request>>(request);
```

Comment:
Could we capture the fact that the error is due to a cycle, somewhere in the naming?

Reply:
👍